File: pytools/examples/golang/src/server/proto/req_test_send.go
package proto

import (
	"packet"
)

type ReqTestSend struct {
	idU8                     uint8
	roleBase                 *MsgRoleBase
	idF32                    []float32
	idOpU8Flag               uint8
	idOpU8                   uint8
	opRoleBaseFlag           uint8
	opRoleBase               *MsgRoleBase
}

func ReqTestSendDecode(pack *packet.Packet) *ReqTestSend {
	reqTestSend := &ReqTestSend{}

	reqTestSend.idU8 = pack.ReadUint8()
	reqTestSend.roleBase = MsgRoleBaseDecode(pack)
	idF32Count := pack.ReadUint16()
	for ;idF32Count > 0; idF32Count-- {
		reqTestSend.idF32 = append(reqTestSend.idF32, pack.ReadFloat32())
	}
	reqTestSend.idOpU8Flag = pack.ReadUint8()
	if reqTestSend.idOpU8Flag == 1 {
		reqTestSend.idOpU8 = pack.ReadUint8()
	}
	reqTestSend.opRoleBaseFlag = pack.ReadUint8()
	if reqTestSend.opRoleBaseFlag == 1 {
		reqTestSend.opRoleBase = MsgRoleBaseDecode(pack)
	}
	return reqTestSend
}

func (this *ReqTestSend) GetIdU8() uint8 {
	return this.idU8
}

func (this *ReqTestSend) GetRoleBase() *MsgRoleBase {
	return this.roleBase
}

func (this *ReqTestSend) GetIdF32() []float32 {
	return this.idF32
}

func (this *ReqTestSend) GetIdOpU8() uint8 {
	return this.idOpU8
}

func (this *ReqTestSend) GetOpRoleBase() *MsgRoleBase {
	return this.opRoleBase
}

File: pytools/data/data.code.java.server/ReqTestSend.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class ReqTestSend
{
	private byte id_u8;
	private MsgRoleBase role_base;
	private List<Float> id_f32 = new ArrayList<Float>();
	private byte id_op_u8_flag;
	private byte id_op_u8;
	private byte op_role_base_flag;
	private MsgRoleBase op_role_base;


	public ReqTestSend(Packet packet)
	{
		id_u8 = packet.readByte();
		role_base = new MsgRoleBase(packet);
		int id_f32_count = PacketUtil.readUShort(packet.readShort());
		for (int i = 0; i < id_f32_count; i++)
		{
			id_f32.add(packet.readFloat());
		}
		id_op_u8_flag = packet.readByte();
		if (id_op_u8_flag == 1)
		{
			id_op_u8 = packet.readByte();
		}
		op_role_base_flag = packet.readByte();
		if (op_role_base_flag == 1)
		{
			op_role_base = new MsgRoleBase(packet);
		}
	}



	public int getIdU8()
	{
		return PacketUtil.readUByte(this.id_u8);
	}

	public MsgRoleBase getRoleBase()
	{
		return this.role_base;
	}

	public List<Float> getIdF32()
	{
		return this.id_f32;
	}

	public int getIdOpU8()
	{
		return PacketUtil.readUByte(this.id_op_u8);
	}

	public MsgRoleBase getOpRoleBase()
	{
		return this.op_role_base;
	}

}
